src/loaders/ncbi_downloader/ncbi_downloader.py

```diff
@@ -34,18 +110,92 @@ def _download_genome_file(download_dir: str, gene_id: str, target_file_ext: list
                 for gene_file_name in gene_file_list:
                     if any([gene_file_name.endswith(ext) for ext in target_file_ext]):
                         result_file_path = os.path.join(download_dir, gene_file_name)
-                        host.download(gene_file_name, result_file_path)
+                        if overwrite or not os.path.exists(result_file_path):
+                            host.download(gene_file_name, result_file_path)
 
             success = True
         except Exception as e:
-            print(e)
+            print(f'Error:\n{e}\nfrom attempt {attempts + 1}.\nTrying to rerun.')
             attempts += 1
 
     if not success:
         raise ValueError(f'Download Failed for {gene_id} after {max_attempts} attempts!')
 
 
-def download_genome_files(gene_ids: list[str], target_file_ext: list[str], result_dir='ncbi_genomes') -> list[str]:
+def _form_gtdb_taxonomy_file_url(release_ver):
+    # form GTDB taxonomy URL for specific GTDB version.
+    # e.g. https://data.gtdb.ecogenomic.org/releases/release207/207.0/ar53_taxonomy_r207.tsv
+
+    if release_ver not in GTDB_RELEASE_VER:
+        raise ValueError(f'Unsupported GTDB release version: {release_ver}')
+
+    file_dir_url = GTDB_DOMAIN + f'release{release_ver}/' + f'{release_ver}.0/'
```

Comment:
Hmm, we really should support dot releases... but the only release with dot releases is 86, and the chance of use using that is very low. How much work would it be to support them?

Reply:
another thing that could be done in a different PR maybe

Reply:
added `todo` comments. I think we will probably never use 86.

Reply:
:+1: